src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2927,7 +2932,7 @@ private ImmutableArray<Diagnostic> GetDiagnosticsForMethodBodiesInTree(SyntaxTre
                 {
                     Debug.Assert(reportUnusedUsings);
 
-                    var discarded = new BindingDiagnosticBag(DiagnosticBag.GetInstance());
+                    var discarded = BindingDiagnosticBag.GetInstance(withDiagnostics: true, withDependencies: false);
```

Comment:
nit: could we use `BindingDiagnosticBag.Discarded`?

Reply:
> could we use BindingDiagnosticBag.Discarded?

This investigation is outside of scope of this PR. I am preserving the original behavior. My guess is that, even though we drop the diagnostics from the bag, consumers of the bag depend on the fact that it is not null. Like passing it to an API that doesn't expect a null value, etc.